packages/botonic-core/src/plugins.ts

```diff
@@ -26,27 +31,22 @@ export function loadPlugins(plugins: PluginConfig<any>[]): ResolvedPlugins {
   return _plugins
 }
 
-export async function runPlugins(
-  plugins: ResolvedPlugins,
-  mode: PluginMode,
-  input: Input,
-  session: Session,
-  lastRoutePath: RoutePath,
-  response: string | null = null
-): Promise<void> {
+export async function runPlugins({
+  requestCoreContext,
+  mode,
+  response = null,
+}: RunPluginArgs): Promise<void> {
+  const plugins = requestCoreContext.plugins
   for (const key in plugins) {
-    const p = await plugins[key]
+    const plugin: Plugin = await plugins[key]
     try {
-      if (mode === 'pre' && p.pre) {
-        await p.pre({ input, session, lastRoutePath, plugins })
+      if (mode === 'pre' && plugin.pre) {
```

Comment:
maybe we have to ensure pre is a callable function, otherwise, we should throw some kind of error like: plugin misconfigured